third_party/intel/backend/driver.py

```diff
@@ -1,26 +1,74 @@
+import importlib.metadata
 import os
 import hashlib
+import shutil
 import tempfile
 from pathlib import Path
 from triton.runtime.build import _build
 from triton.runtime.cache import get_cache_manager
 from triton.backends.compiler import GPUTarget
 from triton.backends.driver import DriverBase
+from packaging.version import Version
+from packaging.specifiers import SpecifierSet
 
 ze_root = os.getenv("ZE_PATH", default="/usr/local")
 include_dir = [os.path.join(ze_root, "include")]
 
-oneapi_root = os.getenv("ONEAPI_ROOT")
-if oneapi_root:
-    include_dir += [
-        os.path.join(oneapi_root, "compiler/latest/include"),
-        os.path.join(oneapi_root, "compiler/latest/include/sycl")
-    ]
+
+def find_sycl(include_dir: list[str], library_dir: list[str]) -> tuple[list[str], list[str]]:
+    """
+    Looks for the sycl library in known places.
+
+    Arguments:
+      include_dir: list of include directories to pass to compiler.
+      library_dir: list of library directories to pass to compiler.
+
+    Returns:
+      enriched include_dir and library_dir.
+
+    Raises:
+      AssertionError: if library was not found.
+    """
```

Comment:
If you return values, you should not change the implicitly passed values, such side effect is not expected from such a function.

```suggestion
    """
    library_dir = []
    include_dir = include_dir.copy()
```